File: SE_Discipline_project1/loops/factorial.c
#include "loops.h"

/**
* factorial - Calculate the factorial of a number.
* @n: The number for which to calculate the factorial.
*
* Return: The factorial of the given number, or -1 if undefined.
*/

int factorial(int n)
{
int i, res;
if (n < 0)
{
return (-1);
}
if (n == 0 || n == 1)
{
return (1);
}
res = n * (n - 1);
for (i = (n - 2); i > 1; i--)
{
res *= i;
}
return (res);
}
  

File: SE_Discipline_project1/loops_py/factorial.py
#!/usr/bin/python3

"""A Python program that
Calculates the factorial of
Any given number
"""


def factorial(n):
    # factorial of 0 and 1 is 1
    if (n == 0 or n == 1):
          return 1
    res = n * (n - 1)
   # loop will start from (n - 2), and end at 1, and will be decremented by -1 in every iteration 
    for i in range((n - 2), 0, -1):
          res *= i
    return res